### ISSUE TITLE
[$1000] User can't download the production app from playstore

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#11132]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to playstore
2. Download new expensify
3. Open the app and login
4. Check if it has a stg badge at top

## Expected Result:
The app should not have the stg badge and stg features

## Actual Result:
 The app is stg

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Android

**Version Number:** 
**Reproducible in staging?:** need repro 
**Reproducible in production?:** need repro
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Upwork job URL:** https://www.upwork.com/jobs/~0138615bf32be2c02c
**Issue reported by:** @Puneet-here 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1662562500746549?thread_ts=1660710162.573049&cid=C01GTK53T8Q

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mallenexpensify** [comment]
@mvtglobally are you able to reproduce?

**Comment by mvtglobally** [comment]
@mallenexpensify @mallenexpensify I was checking with the team on the same. There is a tricky way to download PROD build on Android with the same device as you have staging. Currently we all see STG
Since all our testers are already getting BETA (STG) builds, not sure this result is accurate. 
Is there any chance you can ask someone in Expensify internally who DOES NOT have staging build to download PROD build? 


https://user-images.githubusercontent.com/43995119/196693331-402beb99-5498-4497-b2a6-ca15d1eaeb11.mp4

**Comment by mallenexpensify** [comment]
Asking internally https://expensify.slack.com/archives/C01SKUP7QR0/p1666291488048139

**Comment by mallenexpensify** [comment]
Gonna close cuz @alex-mechler was able to help test and it looks like its not repro
![image](https://user-images.githubusercontent.com/22508304/197034375-1c66a123-4001-4746-b6f6-85a71119d953.png)

![image](https://user-images.githubusercontent.com/22508304/197034384-201d6e39-c78a-4f77-a321-d71c81526b69.png)

**Comment by Puneet-here** [comment]
@mallenexpensify, in the screenshot I can see tha stg badge at the login page.

**Comment by mallenexpensify** [comment]
ah shit... you're right @Puneet-here , great 👀 .  
Reopening and assigning an engineer to take a look

**Comment by iwiznia** [comment]
Oh, I think this is a dupe of https://github.com/Expensify/Expensify/issues/237206 and it is probably internal

**Comment by iwiznia** [comment]
Oh wait, maybe not. Forget that.

**Comment by iwiznia** [comment]
Ummm, isn't play store downloading the staging app because you are in the beta???

**Comment by Puneet-here** [comment]
> Ummm, isn't play store downloading the staging app because you are in the beta???

No, I don't think this is the case because external contributors aren't in the beta and I have tried some different accounts that I've never used at expensify but I still couldn't download the production app.

**Comment by mallenexpensify** [comment]
Job posted https://www.upwork.com/jobs/~0138615bf32be2c02c

**Comment by mayocube** [comment]
Hi
I saw the job on Upwork and did some digging, checked the code and everything seems to be good, unless someone pushed a staging build to the production? 

Coz the environment variables are the only things that differentiates things in the process, so maybe the build pushed to the production (public) is a staging environment build?

If not, can I more context on it?
I mean like access to things like developer console, from which I can find answers and most probably solution?

**Comment by thesahindia** [comment]
> did some digging, checked the code and everything seems to be good, unless someone pushed a staging build to the production?

I also feel the same. @srikarparsi, can you please look into it?

**Comment by smrutiparida** [comment]
I am not too sure but seems to me that it could be a caching issue with what build gets downloaded and then that build does not match with version in package.json in the build. And then we see a badge. I hypothesize this as I see version shown in screenshot is different from what should have been in playstore at that time ( by looking at the release in github). We have new build versions deployed to PROD when this code was being discussed. See the below comment.

<img width="782" alt="Screen Shot 2022-10-22 at 21 35 21" src="https://user-images.githubusercontent.com/5945400/197349484-6a6b461f-ae4f-42c2-ae8d-71560abdd2cf.png">

I could be totally wrong in my many assumptions. Silly question with that,

Why the present android version in playstore is 1.2.16-5 while the github release says it should be 1.2.17-4
<img width="441" alt="Screen Shot 2022-10-22 at 21 46 46" src="https://user-images.githubusercontent.com/5945400/197350007-ebcb1232-7a0d-4be7-a683-2e587026a2df.png">

**Comment by Julesssss** [comment]
I don't believe there was an accidental deploy, as our internal releases line up with the PlayStore builds. Instead, it seems like a regression in our versioning logic as was suggested above. 

Staging version: `1.2.18`
Prod version: `1.2.19-1`

**Comment by mountiny** [comment]
that means there would have to be a PR recently which would break this logic, that could help narrowing this down

**Comment by Julesssss** [comment]
> @mallenexpensify, in the screenshot I can see the stg badge at the login page.

Also, the version is definitely incorrect in [that screenshot](https://github.com/Expensify/App/issues/11132#issuecomment-1286002131).

**Comment by Julesssss** [comment]
Adding the deployBlockerLabel  to prevent further deploys until we can confirm the cause. I know it's useless given this exists on prod, but it'll hopefully speed up the release of the evental fix 🤞

**Comment by Julesssss** [comment]
My current guess is that the bug is in our CI deploy logic, as presumably Applause would have noticed the incorrect version during testing -- or would have reported bugs against a clearly outdated version

**Comment by Puneet-here** [comment]
> that means there would have to be a PR recently which would break this logic, that could help narrowing this down

Just want to inform that I had reported this on Aug 17, if it helps.

**Comment by Julesssss** [comment]
[This](https://github.com/Expensify/App/issues/11880#issue-1410003257) is the deploy checklist for the current prod build, which occurred on August 17th. There were two releases in quick succession on Aug 16th too ([1](https://github.com/Expensify/App/issues/11793) + [2](https://github.com/Expensify/App/issues/11830)). 

Nothing stands out as a potential cause 😕

**Comment by chiragsalian** [comment]
Removing deploy blocker label since its not really a blocker. But keeping the hourly tag since it sounds urgent to address. I think we can also create a fire-room to discuss this.

**Comment by mallenexpensify** [comment]
Any update here @chiragsalian ?  Asking cuz it's hourly and I'm rarely assigned to hourly issues

**Comment by chiragsalian** [comment]
No idea, i haven't investigated this issue since I thought others were investigating. Could either @Julesssss, @Puneet-here, or @thesahindia summarize where we currently stand on the issue and if there are any next steps to tackle this?

**Comment by Julesssss** [comment]
@chiragsalian I wasn't assigned either, just tried to help investigate when I saw it raised. As it's no longer a blocker I think we just need to retest on later releases to see if the staging icon is still shown. 

@Expensify/applauseleads are you still able to reproduce this?

**Comment by thesahindia** [comment]
This is still reproducible but not sure what the problem is.

**Comment by mallenexpensify** [comment]
1. Does this still need to be `Hourly`?
2. Why hasn't it gone overdue if it's `Hourly`?

**Comment by srikarparsi** [comment]
@chiragsalian or @Julesssss do you guys have the bandwith to take over this issue? I'm assigned currently but don't know enough about deploys to troubleshoot. I'll shadow whoever takes it over though so I can learn a little about the process.

**Comment by Julesssss** [comment]
>do you guys have the bandwith to take over this issue? I'm assigned currently but don't know enough about deploys to troubleshoot. 

I'm afraid not, but also I no longer believe this was a deploy issue. Pretty sure there has been a regression at some point

**Comment by mallenexpensify** [comment]
Doubled price to $500 
https://www.upwork.com/jobs/~0138615bf32be2c02c

@michaelhaxhiu Reassigning as I'm heading OOO for a month

**Comment by michaelhaxhiu** [comment]
Ok I've read through this thread. @srikarparsi if you are looking to pass this off, can you post in a slack channel and link here? Let's get this handed off today if we can to keep progress on the `daily` nature.

**Comment by srikarparsi** [comment]
sounds good, thanks @michaelhaxhiu. I just posted [here](https://expensify.slack.com/archives/C03TQ48KC/p1668454666959519)

**Comment by michaelhaxhiu** [comment]
The verdict from the thread that @srikarparsi started is that this will remain on [hold] until the following is finalized:

>Maybe not dupe, but similar in the sense that env doesn't exactly work the same on native. So once this is fixed via this PR: https://github.com/Expensify/App/pull/12546
>It should be easier to investigate or fix

**Comment by smrutiparida** [comment]
> > Maybe not dupe, but similar in the sense that env doesn't exactly work the same on native. So once this is fixed via this PR: #12546. It should be easier to investigate or fix

The environment value that set the badge to STG, is already being computed differently for native.  Which is by comparing pkg.version in the build with android play store set version. Only when both the version are set different, we shall see the STG badge. We must think if there is a process deficiency during prod release or any other manual error that can cause such a thing !

**Comment by michaelhaxhiu** [comment]
This is on Hold, but I'm re-assigning as I'm going OOO. PR is linked above that's blocking this, I have a feeling it could finish sometime next week @maddylewis

**Comment by maddylewis** [comment]
>The verdict from the thread that @srikarparsi started is that this will remain on [hold] until the following is finalized:
>Maybe not dupe, but similar in the sense that env doesn't exactly work the same on native. So once this is fixed via this PR: https://github.com/Expensify/App/pull/12546
It should be easier to investigate or fix

moving to weekly until this is off hold.

**Comment by maddylewis** [comment]
@srikarparsi - could you confirm the details in @tjferriss' comment - https://github.com/Expensify/App/issues/11132#issuecomment-1325483442?

thanks!

**Comment by srikarparsi** [comment]
hey sorry yeah I was waiting on the hold issue to complete before responding. I can take this on internally so I will add the internal label :)

**Comment by srikarparsi** [comment]
I was looking into it today and it seems like the value that points to the staging API just needs to be set to false which will be done in this [PR](https://github.com/Expensify/App/pull/13029). @mountiny could you confirm? If so, I think we can close this issue.

**Comment by mountiny** [comment]
I am not sure if that PR will resolve this. The PR is mainly looking at the API requests. Here the issue is the Staging badge, which in native apps has different logic that which is touched in the aforementioned PR. We use this [getEnvironment method](https://github.com/Expensify/App/blob/bcdb1413fab510ee97c38928aba053b37b1d46d9/src/libs/Environment/getEnvironment/index.native.js#L1-L35) which in android uses this [betachecker api request](https://github.com/Expensify/App/blob/01e7a70a6ae3573c4b04a394c16c9e088cdd59e6/src/libs/Environment/betaChecker/index.android.js#L1-L35) to find out if we match the staging App store version or not.

So I would look at that file/ code and see if by any chance something changed with the version logic there.

**Comment by trjExpensify** [comment]
@srikarparsi have you had a chance to look into this? 

>So I would look at that file/ code and see if by any chance something changed with the version logic there.

**Comment by srikarparsi** [comment]
hey @tjferriss! I haven't had a chance to look into it yet but I'm looking into it now.

**Comment by trjExpensify** [comment]
Awesome, @srikarparsi. This is now one of the oldest bugs in the repo, so it would be great to keep the pace on it. Thanks!

**Comment by srikarparsi** [comment]
Talking to vit about this but believe that the issue is from line 14 in this [PR](https://github.com/Expensify/App/pull/9960)

**Comment by mallenexpensify** [comment]
@maddylewis taking this back, it was assigned to me before I went OOO. 
Let's keep this daily since it sounds like @srikarparsi and @mountiny found the root cause.

**Comment by srikarparsi** [comment]
asked for example of play store url text [here](https://expensify.slack.com/archives/C03TQ48KC/p1670969369651429) to help me write regex.

**Comment by srikarparsi** [comment]
Discussing a more long term solution [here](https://expensify.slack.com/archives/C03TQ48KC/p1670969369651429)

**Comment by srikarparsi** [comment]
Unassigning myself as andrew is working on the fix

**Comment by AndrewGable** [comment]
PR sent 📤

**Comment by mallenexpensify** [comment]
This will hit the 'one week after deployed to production' milestone tomorrow.  It looks like the only person dude compensation is @Puneet-here for reporting.  If I'm missing someone, please comment

**Comment by mountiny** [comment]
Asked in [the PR](https://github.com/Expensify/App/pull/13698#issuecomment-1367624072) if we had closed to loop on the ProductionQA actually.

**Comment by AndrewGable** [comment]
@mallenexpensify Sounds good 👍 - [Testing passed](https://github.com/Expensify/App/pull/13698#issuecomment-1368009296)

**Comment by mallenexpensify** [comment]
@Puneet-here can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~019d52081ca835b9ef

**Comment by Puneet-here** [comment]
Accepted @mallenexpensify

**Comment by mallenexpensify** [comment]
@Puneet-here paid $250 for reporting 
@AndrewGable we want to update regression test steps for this, right?   If so I'll leave open and tackle the last piece. 

BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] The PR that introduced the bug has been identified. Link to the PR:
- [ ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:

**Comment by AndrewGable** [comment]
Yep - We can add a regression test for this.

**Comment by mallenexpensify** [comment]
Posted regression test steps in #bug-zero
https://expensify.slack.com/archives/C01SKUP7QR0/p1672711388472929

**Comment by mallenexpensify** [comment]
Likely running with this for regression test steps, will create the GH for tomorrow 

1. Ensure App isn't loaded on device
2. Click Android on an Android device
3. Follow steps to download app
4. Verify app has downloaded to device
5. Verify the app does not way staging on the login page.

I put 'does not' cuz that should be the expected behavior and staging should be the edge or test case, right?  I'm still new to these steps, I'm assuming we're always creating them for the production version of the app, let me know if that's not correct

**Comment by mallenexpensify** [comment]
Issue created to update test steps https://github.com/Expensify/Expensify/issues/253420

Are these needed before we close this out @AndrewGable ? 

- The PR that introduced the bug has been identified. Link to the PR:
-  The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
-  A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

**Comment by AndrewGable** [comment]
I do not think this is required, the code relied on an external site that we do not have control over, so it actually started to fail without a code change on our side.

**Comment by mallenexpensify** [comment]
Thanks @AndrewGable.  
@thesahindia are you able to complete the above steps?  I think the process we're working towards is to have C+ complete all items they can. For any that are related to backend changes, the C+ would comment on the issue, tag the assigned internal engineer, and provide reasoning why they're unable to complete the steps.

**Comment by thesahindia** [comment]
I don't have much knowledge on this issue since I didn't do anything here. Just a heads-up that @srikarparsi [mentioned](https://github.com/Expensify/App/issues/11132#issuecomment-1346183682) that it was a regression from https://github.com/Expensify/App/pull/9960

**Comment by mountiny** [comment]
I think as Andrew mentioned, @mallenexpensify we should be good to close this one out, seems like the Google API response has changed and that introduced this problem. We could not do much else other than have a regression test which will check for this.

**Comment by mallenexpensify** [comment]
K, closing.  Thanks @thesahindia and @mountiny
